src/admin/config_schema.py

```
@@ -0,0 +1,169 @@
+def load_config(config_path: str) -> KnarrConfig:
+    """Load the index config and all referenced community configs."""
+    config_dir = Path(config_path).parent
+    with open(config_path) as f:
+        index_data = yaml.safe_load(f)
+
+    def community_loader(rel_path: str) -> dict:
+        full_path = config_dir / rel_path
+        if not full_path.exists():
+            full_path = config_dir.parent / rel_path
+        if not full_path.exists():
+            raise ConfigError(f"Community config not found: {rel_path}")
```

Ruff (0.15.11)

[warning] Avoid specifying long messages outside the exception class

(TRY003)